### DISCUSSION TITLE
Portrait orientation for touch

### DISCUSSION THREAD

**Discussion by nradina** [Bodmer/TFT_eSPI#2293] [Q&A - Touch]
Hi everyone,

I just started working with this excellent library and I'm impressed! Great work!

I've been trying out the Squareline Studio/LVGL solution which offers direct compatibility with the TFT_eSPI library.

I created a **240w by 320h** design which displays great, yet the touch coordinates seem to be incorrect. The screen DOES react yet the touch parameters seem to be set for a landscape or another mode.

Could this be a setting I don't have correct within this code?

```
void my_touchpad_read( lv_indev_drv_t * indev_driver, lv_indev_data_t * data )
{
    uint16_t touchX = 0, touchY = 0;
    bool touched = tft.getTouch( &touchX, &touchY, 600 );
```

Curious if I need to alter the **touchX** and **touchY** values?
Also, what the _"600"_ number designate.

Thank you for any help!

**Comment by Bodmer** [comment]
You need to calibrate the screen in the rotation you will use. To do this run the Touch_calibrate sketch with the serial monitor window open and set to 115200 baud. When the calibration is complete the calibration parameters will be sent to the serial monitor window.

You then need to cut and paste those parameters into the lvgl sketch in the setup() section and call the setTouch calibrate function. For example:
```
    tft.begin();
    tft.setRotation( 0 );
    uint16_t calData[5] = { 285, 3593, 227, 3714, 4 };
    tft.setTouch(calData);
```
Then touches will then correspond to the correct area of the screen.

Different screens have different touch threshold values. the 600 is the threshold value, pressure values above this identify a touch event. Typically the low quality resisitance based screens are less sensitive at the corners so the value may need to be adjusted. You can see the raw (uncalibrated) x,y and touch pressure (z value) if you run the TFT_eSPI library diagnostic example "Test_Touch_Controller" and look at the serial monitor window output. Check the sensitivity in ALL corners and set the threshold just below the minimum. Too low and noise will trigger a false touch.

**Reply by nradina** [reply]
Thank you, Bodmer! This is Incredibly helpful. 
What a wonderful library you've created here! Greatful!

**Reply by Bodmer** [reply]
This topic encouraged me to download a copy of Squareline Studio and try it.  I was surprised it produced templates specifically for TFT_eSPI!

The Windows versions had problems with selecting things, I had to click some distance below a widget icon to select it! There was also some issues trying to re-size widgets.

I did manage to produce a very simple GUI in the end with the touch working with an ESP32. Trying to use an RP2040 failed though, this appeared to be due to some incompatibilites in the lvgl core library.

It looks like a promising and useful tool, I will have a look again at some point in the future when some of these bugs are ironed out.

**Reply by nradina** [reply]
Nice. So far it's working great for me thanks to your library!
